### PR TITLE
Feature/oauth2

### PR DESCRIPTION
**Pull request by totorototo**
@guijesuis cadeau!

### CODE REVIEW

File: app/components/login/Login.js [hunk 1]
```diff
@@ -1,42 +1,34 @@
...
-        Linking.addEventListener('url', this.handleOpenURL);
-
-        Linking.canOpenURL(stravaUrl).then(supported => {
-            if (supported) {
-                Linking.openURL(stravaUrl);
-            } else {
-                console.log('Don\'t know how to open URI: ' + this.props.url);
+    handleNavigationChange(event){
+        if(this.refs != null){
+            if (event.url.indexOf('strava://localhost') === -1) {
```
Comment (guijesuis) [inline app/components/login/Login.js]: use a ref callBack https://facebook.github.io/react/docs/refs-and-the-dom.html#the-ref-callback-attribute

File: app/components/login/Login.js [hunk 2]
```diff
@@ -1,19 +1,37 @@
...
+        // }
+        return true;
+    };
+
     render() {
 
-        const{login} = this.props;
+       return (
+            <WebView
+                ref={WEBVIEW_REF}
```
Comment (guijesuis) [inline app/components/login/Login.js]: https://facebook.github.io/react/docs/refs-and-the-dom.html#the-ref-callback-attribute

File: app/helpers/api.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+
+
+export const callApi = ({endpoint, headers, body}) => {
```
Comment (guijesuis) [inline app/helpers/api.js]: may be add a settable default request

Comment (guijesuis) [reply app/helpers/api.js]: rename callJSONApi ?

File: app/index.js [hunk 4]
```diff
@@ -1,37 +1,63 @@
 import React, {Component} from 'react';
+import{Linking} from 'react-native';
 
 import configureStore from './store/configureStore';
 
+import * as login from './actions/login'
```
Comment (guijesuis) [inline app/index.js]: no import * (can't be staticly analysed for lighter js )

File: app/reducers/athlete.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+//constants
+import * as types from './../constants/actionTypes'
```
Comment (guijesuis) [inline app/reducers/athlete.js]: no import *

File: app/reducers/login.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+//constants
+import * as types from './../constants/actionTypes'
```
Comment (guijesuis) [inline app/reducers/login.js]: same here

File: app/sagas/index.js [hunk 7]
```diff
@@ -1,5 +1,5 @@
 //redux-saga
-import { takeEvery, fork } from 'redux-saga/effects'
+import { fork } from 'redux-saga/effects'
 
 //sagas
 import * as authentication from './login'
```
Comment (guijesuis) [inline app/sagas/index.js]: and here again

File: app/services/login.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+//constants
+import * as restConstants from '../constants/rest'
+
```
Comment (guijesuis) [inline app/services/login.js]: and here again

File: app/index.js [hunk 9]
```diff
@@ -1,37 +1,63 @@
 import React, {Component} from 'react';
+import{Linking} from 'react-native';
```
Comment (guijesuis) [inline app/index.js:3]: u need coherent spacing, set it up correctely plz, if not automated, automate it

File: app/sagas/login.js [hunk 10]
```diff
@@ -1,19 +1,80 @@
...
+                    userSignedOut = true;
+                    token = null;
+
+                    yield call(logout);
+                }
+            }
+        }
     }
-    catch(error){
+    catch (error) {
```
Comment (guijesuis) [inline app/sagas/login.js:81]: dosomething here don't silently catch errors -> bad practice -> u will forget it, may be

File: app/services/login.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+export const fetchToken = (form_data) => {
+
+    let headers = new Headers();
+    headers.append('Accept', 'application/json');
+    headers.append('Origin', '*');
+    headers.append('Cache-Control', 'no-cache');
+    headers.append('Content-Type', restConstants.APPLICATION_TYPE.FORM_DATA);
+
```
Comment (guijesuis) [inline app/services/login.js]: may be a helper to build header?

File: package.json [hunk 12]
```diff
@@ -11,6 +11,7 @@
     "react": "~15.4.0-rc.4",
     "react-native": "0.40.0",
     "react-native-router-flux": "^3.37.0",
+    "react-navigation": "^1.0.0-beta.1",
```
Comment (guijesuis) [inline package.json]: is it better than react router V4 ? what do you think?

**Review by guijesuis — CHANGES_REQUESTED** [review]
Plz do some changes in your code

File: app/index.js (hunk 9 above)
Comment (totorototo) [reply app/index.js:3]: rules will be set asap.

**Review by guijesuis — APPROVED** [review]

**Pull request merged**